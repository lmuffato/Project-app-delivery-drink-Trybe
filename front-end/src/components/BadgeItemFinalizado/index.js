import React from 'react';
import PropTypes from 'prop-types';

function BadgeItemFinalizado({ id, descricao, quantidade, valorUnitario }) {
  return (
    <div
      style={ {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        width: '90%',
      } }
    >
      <div
        style={ {
          backgroundColor: 'green',
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
          color: 'black',
          fontWeight: 'bold',
          fontSize: '20px',
          width: '10%',
          borderStartStartRadius: '10px',
          borderEndStartRadius: '10px',
        } }
      >
        <p>{ id }</p>
      </div>
      <div
        style={ {
          backgroundColor: 'blue',
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
          color: 'black',
          fontWeight: 'bold',
          fontSize: '20px',
          width: '30%',
        } }
      >
        <p>{ descricao }</p>
      </div>
      <div
        style={ {
          backgroundColor: 'purple',
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
          color: 'white',
          fontWeight: 'bold',
          fontSize: '20px',
          width: '15%',
        } }
      >
        <p>{ quantidade }</p>
      </div>
      <div
        style={ {
          backgroundColor: 'red',
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
          color: 'white',
          fontWeight: 'bold',
          fontSize: '20px',
          width: '15%',
        } }
      >
        <p>{ `R$ ${valorUnitario.toFixed(2)}` }</p>
      </div>
      <div
        style={ {
          backgroundColor: 'yellow',
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
          color: 'white',
          fontWeight: 'bold',
          fontSize: '20px',
          width: '15%',
        } }
      >
        <p>{ `R$ ${(quantidade * valorUnitario).toFixed(2)}` }</p>
      </div>
      <div
        style={ {
          backgroundColor: 'green',
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
          fontWeight: 'bold',
          fontSize: '20px',
          width: '20%',
          borderTopRightRadius: '10px',
          borderBottomRightRadius: '10px',
        } }
      >
        <button
          type="button"
          onClick={ () => console.log('Teste') }
          style={ {
            border: 'none',
            backgroundColor: 'green',
            color: 'white',
            height: '64px',
            fontSize: '20px',
          } }
        >
          Remover
        </button>
      </div>
    </div>
  );
}

BadgeItemFinalizado.propTypes = {
  id: PropTypes.string.isRequired,
  descricao: PropTypes.string.isRequired,
  quantidade: PropTypes.number.isRequired,
  valorUnitario: PropTypes.number.isRequired,
};

export default BadgeItemFinalizado;
