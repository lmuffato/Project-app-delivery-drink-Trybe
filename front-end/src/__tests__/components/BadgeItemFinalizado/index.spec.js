import React from 'react';
import { render } from '@testing-library/react';
import BadgeItemFinalizado from '../../../components/BadgeItemFinalizado/index';

describe('Snapshot test', () => {
  it('renders correctly', () => {
    const JSON = render(
      <BadgeItemFinalizado
        id="1"
        descricao="Cerveja Brahma"
        quantidade={ 2 }
        valorUnitario={ 4 }
      />,
    );
    expect(JSON).toMatchSnapshot();
  });
});
