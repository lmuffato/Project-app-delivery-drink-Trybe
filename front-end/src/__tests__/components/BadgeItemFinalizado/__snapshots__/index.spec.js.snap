// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="display: flex; justify-content: center; align-items: center; width: 90%;"
      >
        <div
          style="background-color: green; display: flex; justify-content: center; align-items: center; color: black; font-weight: bold; font-size: 20px; width: 10%; border-start-start-radius: 10px; border-end-start-radius: 10px;"
        >
          <p>
            1
          </p>
        </div>
        <div
          style="background-color: blue; display: flex; justify-content: center; align-items: center; color: black; font-weight: bold; font-size: 20px; width: 30%;"
        >
          <p>
            Cerveja Brahma
          </p>
        </div>
        <div
          style="background-color: purple; display: flex; justify-content: center; align-items: center; color: white; font-weight: bold; font-size: 20px; width: 15%;"
        >
          <p>
            2
          </p>
        </div>
        <div
          style="background-color: red; display: flex; justify-content: center; align-items: center; color: white; font-weight: bold; font-size: 20px; width: 15%;"
        >
          <p>
            R$ 4.00
          </p>
        </div>
        <div
          style="background-color: yellow; display: flex; justify-content: center; align-items: center; color: white; font-weight: bold; font-size: 20px; width: 15%;"
        >
          <p>
            R$ 8.00
          </p>
        </div>
        <div
          style="background-color: green; display: flex; justify-content: center; align-items: center; font-weight: bold; font-size: 20px; width: 20%; border-top-right-radius: 10px; border-bottom-right-radius: 10px;"
        >
          <button
            style="background-color: green; color: white; height: 64px; font-size: 20px;"
            type="button"
          >
            Remover
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: flex; justify-content: center; align-items: center; width: 90%;"
    >
      <div
        style="background-color: green; display: flex; justify-content: center; align-items: center; color: black; font-weight: bold; font-size: 20px; width: 10%; border-start-start-radius: 10px; border-end-start-radius: 10px;"
      >
        <p>
          1
        </p>
      </div>
      <div
        style="background-color: blue; display: flex; justify-content: center; align-items: center; color: black; font-weight: bold; font-size: 20px; width: 30%;"
      >
        <p>
          Cerveja Brahma
        </p>
      </div>
      <div
        style="background-color: purple; display: flex; justify-content: center; align-items: center; color: white; font-weight: bold; font-size: 20px; width: 15%;"
      >
        <p>
          2
        </p>
      </div>
      <div
        style="background-color: red; display: flex; justify-content: center; align-items: center; color: white; font-weight: bold; font-size: 20px; width: 15%;"
      >
        <p>
          R$ 4.00
        </p>
      </div>
      <div
        style="background-color: yellow; display: flex; justify-content: center; align-items: center; color: white; font-weight: bold; font-size: 20px; width: 15%;"
      >
        <p>
          R$ 8.00
        </p>
      </div>
      <div
        style="background-color: green; display: flex; justify-content: center; align-items: center; font-weight: bold; font-size: 20px; width: 20%; border-top-right-radius: 10px; border-bottom-right-radius: 10px;"
      >
        <button
          style="background-color: green; color: white; height: 64px; font-size: 20px;"
          type="button"
        >
          Remover
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
